src/engine/database/cosmos.rs

```diff
@@ -0,0 +1,922 @@
+//! Provides database interface types and functions for Cosmos DB
+
+use super::{env_string, env_u16, DatabaseEndpoint, DatabasePool, Transaction};
+use crate::engine::context::{GlobalContext, RequestContext};
+use crate::engine::objects::{Node, NodeRef, Rel};
+use crate::engine::schema::{Info, NodeType};
+use crate::engine::value::Value;
+use crate::Error;
+use async_trait::async_trait;
+use gremlin_client::{
+    ConnectionOptions, GKey, GValue, GraphSON, GremlinClient, Map, ToGValue, VertexProperty,
+};
+use log::trace;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::fmt::Debug;
+
+/// A Cosmos DB endpoint collects the information necessary to generate a connection string and
+/// build a database connection pool.
+///
+/// # Examples
+///
+/// ```rust,no_run
+/// # use warpgrapher::Error;
+/// # use warpgrapher::engine::database::cosmos::CosmosEndpoint;
+/// #
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+///     let ce = CosmosEndpoint::from_env()?;
```

Comment:
I just noticed this syntax. What are the `#` inside the example code block?

Reply:
It hides the line from the documentation output, so you can include lines you need to make the doctest compile that don't have instructive value as part of documentation.